pkg/pillar/types/global.go

```diff
@@ -103,312 +107,417 @@ func (gs *GlobalStatus) UpdateItemValuesFromGlobalConfig(gc GlobalConfig) {
-			newgc.NetworkTestBetterInterval, GlobalConfigMinimums.NetworkTestBetterInterval)
-		newgc.NetworkTestBetterInterval = GlobalConfigMinimums.NetworkTestBetterInterval
-	}
+func (config *ConfigItemValueMap) ModifyAgentValue(key AgentSettingKey, agentName string, newValue string) error {
+	if config.AgentSettings == nil {
+		config.AgentSettings = make(map[string]map[string]ConfigItemValue)
+	}
+	if settingMap, ok := config.AgentSettings[agentName]; ok {
+		if _, okVal := settingMap[string(key)]; okVal {
+			settingMap[string(key)] = ConfigItemValue{StringValue: newValue}
+		} else {
+			return fmt.Errorf("unable to modify per-agent setting for agent: %s because setting %s was not found",
+				agentName, string(key))
+		}
+	} else {
+		return fmt.Errorf("unable to modify per-agent setting for agent: %s because agent was not found",
+			agentName)
+	}
+	return nil
+}
 
-	if newgc.StaleConfigTime < GlobalConfigMinimums.StaleConfigTime {
-		log.Warnf("Enforce minimum StaleConfigTime received %d; using %d",
-			newgc.StaleConfigTime, GlobalConfigMinimums.StaleConfigTime)
-		newgc.StaleConfigTime = GlobalConfigMinimums.StaleConfigTime
-	}
-	if newgc.DownloadGCTime < GlobalConfigMinimums.DownloadGCTime {
-		log.Warnf("Enforce minimum DownloadGCTime received %d; using %d",
-			newgc.DownloadGCTime, GlobalConfigMinimums.DownloadGCTime)
-		newgc.DownloadGCTime = GlobalConfigMinimums.DownloadGCTime
-	}
-	if newgc.VdiskGCTime < GlobalConfigMinimums.VdiskGCTime {
-		log.Warnf("Enforce minimum VdiskGCTime received %d; using %d",
-			newgc.VdiskGCTime, GlobalConfigMinimums.VdiskGCTime)
-		newgc.VdiskGCTime = GlobalConfigMinimums.VdiskGCTime
-	}
-	if newgc.DownloadRetryTime < GlobalConfigMinimums.DownloadRetryTime {
-		log.Warnf("Enforce minimum DownloadRetryTime received %d; using %d",
-			newgc.DownloadRetryTime, GlobalConfigMinimums.DownloadRetryTime)
-		newgc.DownloadRetryTime = GlobalConfigMinimums.DownloadRetryTime
-	}
-	if newgc.DomainBootRetryTime < GlobalConfigMinimums.DomainBootRetryTime {
-		log.Warnf("Enforce minimum DomainBootRetryTime received %d; using %d",
-			newgc.DomainBootRetryTime, GlobalConfigMinimums.DomainBootRetryTime)
-		newgc.DomainBootRetryTime = GlobalConfigMinimums.DomainBootRetryTime
-	}
-	if newgc.RktGCGracePeriod < GlobalConfigMinimums.RktGCGracePeriod {
-		log.Warnf("Enforce minimum RktGCGracePeriod received %d; using %d",
-			newgc.RktGCGracePeriod, GlobalConfigMinimums.RktGCGracePeriod)
-		newgc.RktGCGracePeriod = GlobalConfigMinimums.RktGCGracePeriod
-	}
-	if newgc.Dom0MinDiskUsagePercent < GlobalConfigMinimums.Dom0MinDiskUsagePercent {
-		log.Warnf("Enforce minimum Dom0MinDiskUsagePercent received %d; using %d",
-			newgc.Dom0MinDiskUsagePercent, GlobalConfigMinimums.Dom0MinDiskUsagePercent)
-		newgc.Dom0MinDiskUsagePercent = GlobalConfigMinimums.Dom0MinDiskUsagePercent
-	}
-	return newgc
+func ParseSpecToValue(configSpec ConfigItemSpec) ConfigItemValue {
```

Comment:
This function is better suited to be a method of ConfigItemSpec. For a given ItemSpec, it is getting the Default Item..

func (configSpec ConfigItemSpec) DefaultItemValue() ConfigItemValue {
   <u r code inside> 
}